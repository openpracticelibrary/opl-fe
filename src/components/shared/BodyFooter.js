import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import OPLlogo from "../../assets/icons/open-practice-librar.png";
import labsLogo from "../../assets/icons/logo-red-hat-open-innovation-labs-a-black-rgb.png";
import Link from "@material-ui/core/Link";
import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles((theme) => ({
  footer: {
    marginTop: theme.spacing(5),
    flexGrow: 1,
  },
  toolbarLink: {
    padding: theme.spacing(2),
    flexShrink: 0,
  },
  listFont: {
    color: "#2e2e2e",
  }
}
));

const sections = [
  { title: "PRIVACY", url: "#" },
  { title: "TERMS", url: "#" },
  { title: "POLICIES", url: "#" },
  { title: "HELP", url: "#" },
];

export default function BodyFooter() {
  const classes = useStyles();

  return (
    <footer className={classes.footer}>
      <Grid
        container
        direction="row"
        justify="space-between"
        alignItems="baseline"
      >
        <Grid item xs={5}>
          <Grid
            container
            direction="column"
            justify="flex-start"
            alignItems="flex-start"
            spacing={1} 
          >
            <Grid item>
              <img
                src={OPLlogo}
                alt="OPL"         
              />
            </Grid>
            <Grid item>
              <Grid
                container
                direction="row"
                justify="flex-start"
                alignItems="flex-start"
              >
                <Grid item xs={3}>
                  <Typography variant="caption">
                      powered by 
                  </Typography>
                </Grid>
                <Grid item xs={3}>
                  <img
                    src={labsLogo}
                    alt="labs"
                    height="31px"
                    width="162px"
                  />
                </Grid>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={5}>
          <Grid
            container
            direction="column"
            justify="space-between"
            alignItems="flex-end"    
            spacing={2} 
          >
            <Grid item>
              <Typography variant="overline">
                CONNECT 
              </Typography>
            </Grid>
            <Grid item>
              {sections.map((section, i) => (
                <Link
                  key={i}
                  href={section.url}
                  className={classes.toolbarLink}
                >
                  <Typography variant="overline" className={classes.listFont}>
                    {section.title}
                  </Typography>
                </Link>
              ))}
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </footer>
  );
}
