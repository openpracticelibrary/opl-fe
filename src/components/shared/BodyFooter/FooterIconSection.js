import React from "react";
import Link from "@material-ui/core/Link";
import { YoutubeIcon, EmailIcon, PodIcon, InstaIcon, GitIcon } from "../../../assets/icons";
import Grid from "@material-ui/core/Grid";

const sections = [
  { url: "mailto:info@openpracticelibrary.com", icon: <EmailIcon /> },
  { url: "https://podcasts.apple.com/us/podcast/open-practice-podcast/id1501715186", icon: <PodIcon /> },
  { url: "https://www.instagram.com/openpracticelibrary/?hl=en", icon: <InstaIcon /> },
  { url: "https://github.com/openpracticelibrary", icon: <GitIcon /> },
  { url: "https://www.youtube.com/channel/UCBGbB7xVsmf-yQ8NXn7TOVg?view_as=subscriber", icon: <YoutubeIcon /> },
];

export default function FooterIconSection(props) {
  return (
    <>
      <Grid
        container
        direction="row"
        spacing={1}
        alignItems="center"
      >
        {sections.map((section, i) => (
          <Grid item key={i}>
            <Link href={section.url} target="_blank">
              {section.icon}
            </Link>
          </Grid>
        ))}
      </Grid>
    </>
  );
}
