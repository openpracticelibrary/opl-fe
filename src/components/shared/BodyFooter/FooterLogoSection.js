import React from "react";
import { Typography } from "@material-ui/core";
import { OPLLogo } from "../../../assets/icons";
import logoLabs from "../../../assets/images/logo-red-hat-open-innovation-labs-a-standard-rgb.png";
import Grid from "@material-ui/core/Grid";

export default function FooterLogoSection() {
  return (
    <>
      <Grid
        container
        direction="column"
        justify="flex-start"
        alignItems="center"
      >
        <Grid item>
          <OPLLogo height="37px" width="221px" />
        </Grid>
        <Grid item>
          <Grid
            container
            direction="row"
            justify="flex-start"
            alignItems="flex-start"
          >
            <Grid item xs={6}>
              <Typography variant="overline" data-testid="poweredBy">
                {`powered by`}
              </Typography>
            </Grid>
            <Grid item xs={6}>
              <img
                src={logoLabs}
                alt="Open Innovation Labs"
                height="31px"
                width="162px"
              />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </>
  );
}
