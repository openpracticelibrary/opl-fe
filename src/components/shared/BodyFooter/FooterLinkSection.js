import React from "react";
import { Typography } from "@material-ui/core";
import Grid from "@material-ui/core/Grid";
import { makeStyles } from "@material-ui/core/styles";
import FooterIconSection from "./FooterIconSection";
//import Link from "@material-ui/core/Link";
import { navigate } from "@reach/router";

const useStyles = makeStyles((theme) => ({
  space: {
    padding: theme.spacing(2),
  },
  color: {
    fontSize: "0.9rem",
    color: "#6c6c6c",
  },
  connect: {
    color: theme.palette.common.black,
    fontWeight: "500",
  },
}));

const sections = [
  { title: "Practices", url: "/practice" },
  { title: "About", url: "#" },
  { title: "Latest", url: "#" },
  { title: "Terms", url: "#" },
  { title: "Help", url: "#" },
];

export default function FooterLogoSection() {
  const classes = useStyles();

  return (
    <>
      <Grid
        container
        direction="column"
        justify="space-between"
        alignItems="flex-end"
        spacing={2}
      >
        <Grid item>
          <Grid
            container
            direction="row"
            spacing={2}
            alignItems="center"
          >
            <Grid item>
              <Typography
                variant="button"
                data-testid="connect"
                className={classes.connect}
              >
                Please connect with us:
              </Typography>
            </Grid>
            <Grid item>
              <FooterIconSection />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid
            container
            direction="row"
            spacing={3}
          >
            {sections.map((section, i) => (
              <Grid item key={i}>
                <Typography
                  variant="button"
                  className={classes.color}
                  onClick={() => navigate(section.url)}
                >
                  {section.title}
                </Typography>
              </Grid>
            ))}
          </Grid>
        </Grid>
      </Grid>
    </>
  );
}
