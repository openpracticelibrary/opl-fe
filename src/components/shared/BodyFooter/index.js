import React from "react";
import {makeStyles} from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import FooterLogoSection from "./FooterLogoSection";
import FooterLinkSection from "./FooterLinkSection";
import Box from '@material-ui/core/Box';
import Divider from "@material-ui/core/Divider";

const useStyles = makeStyles((theme) => ({
  footer: {
    margin: theme.spacing(6, 0)
  },
}));

export default function BodyFooter() {
  const classes = useStyles();

  return (
    <>
      <Divider/>
      <footer className={classes.footer}>
        <Grid container>
          <Grid item xs={2}>
          </Grid>
          <Grid item xs={8}>
            <Box display="flex">
              <Box flexShrink={1}>
                <FooterLogoSection/>
              </Box>
              <Box width="100%">
                <FooterLinkSection/>
              </Box>
            </Box>
          </Grid>
          <Grid item xs={2}>
          </Grid>
        </Grid>
      </footer>
    </>
  );
}
