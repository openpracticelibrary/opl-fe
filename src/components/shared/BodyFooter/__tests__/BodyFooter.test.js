import React from "react";
import { ThemeProvider } from "@chakra-ui/core";
import { render, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer';
import customTheme from '../../../../theme';
import BodyFooter from '../';

afterEach(cleanup);

it("renders BodyFooter correctly", ()=>{
  const { getByTestId } = render(
    <ThemeProvider theme={customTheme}>
      <BodyFooter/>
    </ThemeProvider>
  );
  expect(getByTestId("footer")).toBeInTheDocument();
  expect(getByTestId("connect")).toHaveTextContent("Please connect with us:");
  expect(getByTestId("poweredBy")).toHaveTextContent("powered by");
});

it("matches the snapshot", ()=>{
  const tree = renderer.create(
    <ThemeProvider theme={customTheme}>
      <BodyFooter/>
    </ThemeProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
