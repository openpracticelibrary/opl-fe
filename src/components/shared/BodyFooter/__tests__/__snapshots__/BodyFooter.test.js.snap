// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
Array [
  <hr
    className="MuiDivider-root"
  />,
  <footer
    className="makeStyles-footer-1"
    data-testid="footer"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <div
          className="MuiBox-root MuiBox-root-157"
        >
          <div
            className="MuiBox-root MuiBox-root-158"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <img
                  alt="OPL"
                  src="open-practice-librar.png"
                />
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <div
                  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
                  >
                    <span
                      className="MuiTypography-root MuiTypography-caption"
                    >
                      powered by
                    </span>
                  </div>
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
                  >
                    <img
                      alt="Open Innovation Labs"
                      height="31px"
                      src="logo-red-hat-open-innovation-labs-a-black-rgb.png"
                      width="162px"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-159"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-end MuiGrid-justify-xs-space-between"
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <span
                  className="MuiTypography-root MuiTypography-overline"
                >
                  CONNECT
                </span>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <a
                  className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-space-146 MuiTypography-colorPrimary"
                  href="#"
                  onBlur={[Function]}
                  onFocus={[Function]}
                >
                  <span
                    className="MuiTypography-root makeStyles-color-147 MuiTypography-overline"
                  >
                    PRIVACY
                  </span>
                </a>
                <a
                  className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-space-146 MuiTypography-colorPrimary"
                  href="#"
                  onBlur={[Function]}
                  onFocus={[Function]}
                >
                  <span
                    className="MuiTypography-root makeStyles-color-147 MuiTypography-overline"
                  >
                    TERMS
                  </span>
                </a>
                <a
                  className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-space-146 MuiTypography-colorPrimary"
                  href="#"
                  onBlur={[Function]}
                  onFocus={[Function]}
                >
                  <span
                    className="MuiTypography-root makeStyles-color-147 MuiTypography-overline"
                  >
                    POLICIES
                  </span>
                </a>
                <a
                  className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-space-146 MuiTypography-colorPrimary"
                  href="#"
                  onBlur={[Function]}
                  onFocus={[Function]}
                >
                  <span
                    className="MuiTypography-root makeStyles-color-147 MuiTypography-overline"
                  >
                    HELP
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      />
    </div>
  </footer>,
]
`;
